### ISSUE TITLE
Suggestion for Loading Component 

### ISSUE THREAD

**Issue by Dinuz** [buefy/buefy#862]
Would be possible adding a ref props to the component to be used specifically in a programmatic approach ?

It will be extremely useful in the programmatic case: think on an http submission, it will allow to programmatically close the loading in the response promise callback.

At the moment this is not possible, simply because the promise will not get the variable defined as $loading.open, that will always result undefined.

**Comment by jacurtis** [comment]
I have to agree. This would be very powerful if it could be called programatically from outside the component. Something like `this.$loading.close()`. I am able to call `app.$loading.open()` but not `close()`? That seems odd.

Like @Dinuz i am using it with an http request. I call up the loading screen before the call is made, and then (want to) close it after its complete.

**Comment by Dinuz** [comment]
@jtommy yes I saw it clearly, and I tried that solution before writing here.
Unfortunately, as I mentioned in my first post, that solution is not a workable solution when you have a promise returned by http callback.

The callback return just the response, and there is no way to access the const defined before with the open. The callback is totally and completely blind respect to external declared variables. The only thing that see is "this". Unfortunately without a ref, the this cannot be used to target the loading component, moreover the component when generated programmatically resides outside the app in the dom.

The callback in vue-resource or axios will not allow access to an external variable, that is the issue.
Hope this clarified the issue.

**Comment by vjoao** [comment]
Can't you just use it like this?

```
export default {
  methods: {
    callPromise () {
      const loading = this.$loading.open()
      const call = new Promise((resolve, reject) => {
        setTimeout(() => resolve(true), 2000)
      })

      call.then((value) => {
        loading.close()
      })
    }
  }
}
```

**Comment by Dinuz** [comment]
@vjoa the problem is not the loading itself, the problem is using it inside of an http post and callback. You cannot access the close or the promise returned in ur case in the http callback. 
No I cannot use simply in the way you described, that doesn’t solve the issue.

**Comment by vjoao** [comment]
I use axios, it returns a promise just like I did to explain (with setTimeout to simulate async nature of promises). You must be doing something wrong on your side.

**Comment by Dinuz** [comment]
@vjoao let me show you what I am doing (probably it is some fault on my side), just to put things in perspective:

```
methods: {
    validateBeforeSubmit: function () {
      let vm = this
      this.$validator.validateAll().then(result => {
        if (result) {
          const loadingComponent = this.$loading.open()
          this.$http.post('http://myUrl', data_to_post).then((res) => {
            // HERE IS WHERE I WANT TO CALL THE LOADING CLOSE
           When I call the close it's always undefined.

           // Here I do other things like calling a modal etc
          }, res => {
            // I deal with errors eventually
            }
          })
        }
      })
    }
```

My issue is that I am not able to reference the loading that I open, and If I define in a method as you did, I am still not able to call in that point.

What I am doing wrong?

**Comment by vjoao** [comment]
You are using promises wrongly. I reproduced your mistake: https://jsfiddle.net/ochL02vz/6/
Make sure you study how promises (also async / await) works.

**Comment by Dinuz** [comment]
@vjoa thank you for taking the time to look into this. Tomorrow first thing I do is reviewing my execution. 
However there is a difference between our code, my http is wrap in a promise yours not; could be that the source of all my struggles?
Otherwise on first sight the rest of your code is identical to mine.

**Comment by Dinuz** [comment]
@vjoa I couldn’t wait and I went back to my laptop.
I found my mistake, thanks to you buddy(thank you again for that), everything works fine and great and I will close this issue, the beefy component works great and I am sorry for all this back and forward.

My code was not working due to a mistyping that I couldn’t spot if wasn’t for your code.

My example was cut down extremely, and there are multiple lines in that then. When I looked at it with more attention I saw immediately (thanks to your example) the error, the loading.close was called with a this(that I don’t know how got there in the first place), clearly it was undefined, that is why the code was not working.

I couldn’t find the mistake for the life of me, until I checked the difference among our code. I had a this that was not supposed to be there, and I didn’t realize was there until I saw your script.

Man thank you 2 times today:))

@jtommy sorry for the misunderstanding...gonna close this now.
